plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMigrateProcedure.java

```diff
@@ -224,20 +226,117 @@ public void testMigrateUnsupportedColumnType()
     {
         String tableName = "test_migrate_unsupported_column_type_" + randomNameSuffix();
         String hiveTableName = "hive.tpch." + tableName;
+
+        assertUpdate("CREATE TABLE " + hiveTableName + " AS SELECT tinyint '1' x", 1);
+        assertQuery("SELECT * FROM " + hiveTableName, "SELECT 1");
+
+        assertQueryFails("CALL iceberg.system.migrate('tpch', '%s')".formatted(tableName), "Type not supported for Iceberg: tinyint");
```

Comment:
we should add support for migrating tinyint and smallint (they should get covereted to integer)
add `// TODO (issue link) support tinyint, smallint in migrate()` here.

Reply:
https://github.com/trinodb/trino/issues/17946